crates/matrix-sdk/src/http_client.rs

```diff
@@ -221,32 +206,90 @@ impl HttpClient {
                     }
                 };
 
-                let raw_response = self
+                let response = self
                     .inner
                     .send_request(clone_request(&request), config.timeout)
                     .await
                     .map_err(error_type)?;
 
-                trace!("Got response: {raw_response:?}");
-
-                let response = Request::IncomingResponse::try_from_http_response(raw_response)
-                    .map_err(|e| error_type(HttpError::from(e)))?;
-
                 Ok(response)
             };
 
             retry::<_, HttpError, _, _, _>(backoff, send_request).await?
         };
 
         #[cfg(target_arch = "wasm32")]
-        let response = {
-            let raw_response = self.inner.send_request(request, config.timeout).await?;
-            trace!("Got response: {raw_response:?}");
+        let response = self.inner.send_request(request, config.timeout).await?;
 
-            Request::IncomingResponse::try_from_http_response(raw_response)?
+        Ok(response)
+    }
+
+    #[instrument(
+        skip(self, access_token, config, request, user_id),
+        fields(config, path, user_id, status)
+    )]
+    pub async fn send<R>(
+        &self,
+        request: R,
+        config: Option<RequestConfig>,
+        homeserver: String,
+        access_token: Option<&str>,
+        user_id: Option<&UserId>,
+        server_versions: &[MatrixVersion],
+    ) -> Result<R::IncomingResponse, HttpError>
+    where
+        R: OutgoingRequest + Debug,
+        HttpError: From<FromHttpResponseError<R::EndpointError>>,
+    {
+        let span = tracing::Span::current();
+
+        let config = match config {
+            Some(config) => config,
+            None => self.request_config,
         };
 
-        Ok(response)
+        // Avoid logging the `None`.
+        span.record("config", debug(config));
```

Comment:
If the `None` argument should not be logged, this should be part of the `Some` match arm above. This comment doesn't really makes sense to me.

Reply:
Ohh, so the comment is trying to say "log after obtaining the effective config, to not skip logging if it is the default". I read it as "we want to avoid logging the config if it's the default".

Reply:
Yeah, I though it's useful to show this all the time, since nobody will know what the default is and it aligns all the logs visually so it's easy to skip this anyways.